crates/agentgateway/src/proxy/httpproxy.rs

```diff
@@ -485,6 +481,39 @@ impl HTTPProxy {
 		log.listener_name = Some(selected_listener.name.clone());
 
 		debug!(bind=%bind_name, listener=%selected_listener.key, "selected listener");
+
+		// Get frontend policies for dynamic tracing
+		debug!(gateway=%selected_listener.gateway_name, "Looking up frontend policies");
+		let frontend_policies = inputs
+			.stores
+			.read_binds()
+			.frontend_policies(selected_listener.gateway_name.clone());
+
+		// Debug: Check if tracing policy is found
+		if let Some(tp) = &frontend_policies.tracing {
+			debug!(
+				service_name=%tp.config.service_name,
+				attrs_count=%tp.config.attributes.len(),
+				"Using dynamic tracer from frontend policy"
+			);
+		} else {
+			debug!("No frontend tracing policy found, using static tracer");
+		}
+
+		// Use dynamic tracer from frontend policy if available, otherwise use static tracer
+		if trace_sampled {
+			log.tracer = frontend_policies
+				.tracing
+				.as_ref()
+				.map(|tp| (*tp.tracer).clone())
```

Comment:
Do we need to clone the inner tracer here or can we use `tp.tracer.clone()` and share it? Or is there a reason we want independent tracer instances?